Bento/Views/RequiringPreSizingLayoutPass.swift

```diff
@@ -37,7 +37,13 @@ extension UIView {
 
     func triggerPresizingLayoutPassIfNeeded(forTargetSize size: CGSize) {
         if needsPresizingLayoutPass {
-            bounds.size = size
+            // NOTE: the target size passed in always has a height of zero
+            // when called from `systemLayoutSizeFitting(_ targetSize: CGSize)`
+            // All we care about is the width. Setting the height to zero
```

Comment:
> All we care about is the width.

Is that true it case of UICollectionView? 🤔

Reply:
Well, `systemLayoutSizeFitting(_ targetSize: CGSize)` is only overridden for `TableViewContainerCell` and `TableViewHeaderFooterView`, it's not needed for `UICollectionView`.

However `AnyRenderable.rendered(size: CGSize, inheritedMargins: UIEdgeInsets)` also calls `triggerPresizingLayoutPassIfNeeded`, and this applies to `UICollectionView` too.

However, the whole premise of `PreSizingLayoutPassRequiring` is *height* calculation which is dependent on a known width. From the comments for this protocol:

```swift
/// Declare that the view requires an extra layout pass before Auto Layout item sizing happens, so that height
/// calculation for components containing free-form UI elements, e.g. text labels, would be correct.
```

Also from the comments:

```swift
/// - important: The mechanism is always enabled for Bento-enabled `UITableView`, both vanilla and size-caching. For
///              `UICollectionView` however, it is available only for size-caching `UICollectionView`, since
///              `UICollectionView` can be flexibly laid out and thus Bento does not have sufficient knowledge when
///              working with vanilla `UICollectionView`.
```

Basically, for `UICollectionView` you have to depend on auto-layout, because your cells might have a fixed width *and* height.